packages/shared/src/hooks/companion/useBackgroundRequest.ts

```diff
@@ -14,12 +14,13 @@ export const useBackgroundRequest = (
   const client = useQueryClient();
 
   useRawBackgroundRequest(({ key, ...args }) => {
-    if (!enabled || !isQueryKeySame(key, queryKey)) {
+    const validKey = Array.isArray(key) ? key : [key];
+    if (!enabled || !isQueryKeySame(validKey, queryKey)) {
       return;
     }
 
     if (callback) {
-      callback({ key, ...args });
+      callback({ validKey, ...args });
```

Comment:
should be `key: validKey` since callback expects `key`?

Reply:
I guess so, probably should make it typed eventually, but see it's widely used as generic now.